### PR TITLE
Fix #1765: Replace Arrays.sort implementation.

### LINKED ISSUE #1765: Update java.util.Arrays.sort to the clean implem in Scala 2.11.7
Quoting @adriaanm from https://github.com/scala-js/scala-js/pull/1746#issuecomment-115301872:

> Hey @sjrd, @nicolasstucki, @gzm0: it looks like you copy/pasted code from 2.11.5 which was copy/pasted from the forbidden fruit, so please update to the ones in 2.11.7 ASAP: scala/scala#4557

We must update our implementations of `java.util.Arrays.sort` to the clean ones in Scala 2.11.7.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by  @sjrd and @adriaanm

File: javalib/src/main/scala/java/util/Arrays.scala [hunk 1]
```diff
@@ -80,81 +80,283 @@ object Arrays {
...
+    quickSort[T](a, fromIndex, toIndex)
   }
 
   @inline
   private def sortRangeAnyRefImpl(a: Array[AnyRef], fromIndex: Int, toIndex: Int)(
       implicit ord: Ordering[AnyRef]): Unit = {
-    checkIndicesForCopyOfRange(a.length, fromIndex, toIndex)
-    stableSortRangeAnyRefImpl(a, fromIndex, toIndex - 1, new Array[AnyRef](a.length),
-      (a: AnyRef, b: AnyRef) => ord.lt(a, b))
+        checkIndicesForCopyOfRange(a.length, fromIndex, toIndex)
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/Arrays.scala]: Indentation issue.

File: javalib/src/main/scala/java/util/Arrays.scala [hunk 2]
```diff
@@ -80,81 +80,283 @@ object Arrays {
...
+  private def sortAnyRefImpl(a: Array[AnyRef])(implicit ord: Ordering[AnyRef]): Unit =
+    quickSortAnyRef(a, 0, a.length)
+
+  // Implementation of sorting based on Scala 2.11.7 scala.util.Sorting
+  private final val qSortThreshold = 16
+
+  // Ordering[T] might be slow especially for boxed primitives, so use binary search variant of insertion sort
+  // Caller must pass iN >= i0 or math will fail.  Also, i0 >= 0.
+  @noinline
+  private final def insertionSort[@specialized T](a: Array[T], i0: Int, iN: Int)(implicit ord: Ordering[T]): Unit = {
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/Arrays.scala]: The `insertionSort` helper should be right after the `quickSort` it helps, not before. (Same for the `AnyRef` variant.)

**Comment by sjrd** [comment]
That's all.

**Comment by nicolasstucki** [comment]
Updated

**Comment by sjrd** [comment]
LGTM

**Pull request merged**